python/pc_ble_driver_py/ble_adapter.py

```diff
@@ -327,9 +327,19 @@ def write_cmd(self, conn_handle, uuid, data):
                                            handle,
                                            data,
                                            0)
-        self.driver.ble_gattc_write(conn_handle, write_params)
-        self.evt_sync[conn_handle].wait(evt = BLEEvtID.evt_tx_complete)
 
+        # Send packet and skip waiting for TX-complete event. Try maximum 3 times.
+        for _ in range(3):
+            try:
+                self.driver.ble_gattc_write(conn_handle, write_params)
+                return
+            except NordicSemiException as e:
+                # Retry if BLE_ERROR_NO_TX_PACKETS error code.
+                if "Error code: 12292" in e.message:
```

Comment:
No error code is given from NordicSemiErrorCheckright now, except for the one in the message.
It is possible to change how the function raises exception to give an error code and a message separated in a tuple. Like this:
raise NordicSemiException(err_code, 'Failed to {}.'.format(wrapped.__name__))

I could not find the BLE error in pc_ble_driver_sd_api_v3.py, so we'll have to defined the error code in the ble_driver.py. The code in ble_adapter.py could look like this:

            except NordicSemiException as e:
                # Retry if BLE_ERROR_NO_TX_PACKETS error code.
                if e.args[0] == BLE_ERROR_NO_TX_PACKETS:
                    self.evt_sync[conn_handle].wait(evt=BLEEvtID.evt_tx_complete, timeout=1)
                else:
                    raise e